### ISSUE TITLE
The file in the old version should load when going from compare to the browse page

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-code-manager#1436]
Followup for https://github.com/mozilla/addons-code-manager/issues/1421

STR:
1. Go to https://code.addons-dev.allizom.org/en-US/compare/7560/versions/1252898...1688247/?path=install.rdf
2. Click the Compare Versions button and then click the Browse page for the old version 

Expected result:
the page should load 

Actual result:
Loading linter information...

Notes: 
I receive a 404 in the Network for https://addons-dev.allizom.org/en-US/developers/addon/webmail-ad-blocker/file/124749/validation.json
It is reproducible on dev with FF73(Win10) for other files of these type (which were deleted in the new version). 

![404again](https://user-images.githubusercontent.com/33448286/74542384-2714ba00-4f4c-11ea-9d07-13d7481817ef.png)

**Comment by mirefly** [comment]
Thanks for filing this. When investigating #1327, I noticed the same problem in compare mode ( https://code.addons-dev.allizom.org/en-US/compare/7560/versions/1252898...1510628/). I believe 404 error is because the version is too old and has no linter info, but code-manager should load the file when failed to fetch linter messages (`validtion.json` file).

**Comment by ioanarusiczki** [comment]
@mirefly Yes, further looking into versions, same for https://code.addons-dev.allizom.org/en-US/compare/7560/versions/1509666...1510628/

**Comment by mirefly** [comment]
I could not reproduce this in dev now (d120186). I had no idea how it got fixed... :)

**Comment by bobsilverberg** [comment]
This issue is about the fact that linter information could be missing for a version, and when that happens we don't display the code.

To fix this we should detect when there's a 404 from `validation.json` and still show the code, and also display a banner explaining that we were unable to retrieve validation information.

**Comment by ioanarusiczki** [comment]
Here's another example - I see for this addon a 404 on validation.json in the Network:

https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/noscript?page=1#history

I couldn't load any of the most recent versions uploaded.
